packages/gatsby-plugin-canonical-urls/src/gatsby-browser.js

```diff
@@ -1,4 +1,10 @@
+const url = require(`url`)
+
 exports.onRouteUpdate = ({ location }) => {
   const domElem = document.querySelector(`link[rel='canonical']`)
-  domElem.setAttribute(`href`, `${window.location.origin}${location.pathname}`)
+  var existingValue = domElem.getAttribute(`href`)
+  if ( existingValue ) {
+    const myUrl = url.parse(existingValue)
+    domElem.setAttribute(`href`, `${myUrl.protocol}//${myUrl.host}${location.pathname}${location.search}${location.hash}`)
```

Comment:
Depending on how you’re using query strings or hashes this could result in duplicate content with conflicting canonicals. Perhaps there should be a way to opt-in to using them?

Reply:
Ah, true. Hmm, I would be fine with going back to just using pathname and not adding search or hash for now.

Reply:
I can remove the location.search and the location.hash if you think that will cause issues. We can always revisit in the future. Let me know if you want me to do so.